File: benchmarks/classes/bench.go
package main

import "fmt"

type Person struct {
	age    int
	weight int
}

func (p *Person) getValue() int {
	return p.age + p.weight
}

func main() {
	sum := 0
	for i := 0; i < 1000000; i++ {
		p := Person{age: i, weight: i + 1}
		sum += p.getValue()
	}
	fmt.Println(sum)
}

File: benchmarks/classes/bench.py
class Person:
    def __init__(self, age, weight):
        self.age = age
        self.weight = weight
        
    def get_value(self):
        return self.age + self.weight

def class_test():
    sum_val = 0
    for i in range(1000000):
        p = Person(i, i + 1)
        sum_val += p.get_value()
    return sum_val

if __name__ == "__main__":
    print(class_test())
